lib/phoenix/router.ex

```diff
@@ -293,8 +293,15 @@ defmodule Phoenix.Router do
       end
 
     quote do
-      defp do_call(unquote(conn), opts) do
-        unquote(call)
+      defp do_call(unquote(conn) = conn, opts) do
+        case conn.private[:phoenix_bypass] do
```

Comment:
Please pattern match this on the function head, it is going to be much, much, much faster.